### PR TITLE
EPROD 934 use icrc1 metadata in place of 3 canister calls

### PR DESCRIPTION
**Pull request by veeso**
- **wip**
- **wip**
- **fix: tests**

## Issue ticket

Issue ticket link:

### Code conventions

- [ ] I have performed a self-review of my code
- [ ] Every new function is documented
- [ ] Object names are auto explicative
- [ ] The PR follows the [project conventions](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/23330839/Conventions)
- [ ] The code follows the [style guidelines of the project](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/24444929/Rust+Conventions)
- [ ] The code follows the [project security best practices](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/35225620/Security+Considerations)

### Security 

- [ ] The PR does not break APIs backward compatibility
- [ ] The PR does not break the stable storage backward compatibility

### Testing 

- [ ] Every function is properly unit tested
- [ ] I have added integration tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/icrc2-minter/src/tokens/icrc1.rs [hunk 1]
```diff
@@ -104,20 +113,99 @@ async fn query_icrc1_configuration(token: Principal) -> Result<TokenConfiguratio
...
+/// If it is, we can use the standard ICRC-1 metadata query to get the token info.
+/// Otherwise, we have to use dedicated queries to get the token info.
+async fn query_icrc1_token_info<C>(
+    icrc_client: &IcrcCanisterClient<C>,
+    has_token_info_in_metadata: bool,
+) -> Result<TokenInfo>
+where
+    C: CanisterClient,
+{
+    // If the token info is set in metadata, we can use the standard ICRC-1 metadata query.
```
Comment (blutooth) [inline src/icrc2-minter/src/tokens/icrc1.rs:127]: Can we check which tokens don't have the info set in the metadata, this should be a requirement for all tokens.  In that case, is it necssary to have: query_icrc1_token_info_from_dedicated_queries ?

Comment (veeso) [reply src/icrc2-minter/src/tokens/icrc1.rs:127]: In general I haven't found any ICRC which doesn't expose those keys in the metadata, but the ICRC standard states:

```md
ledger can expose metadata to simplify integration with wallets and improve user experience. The client can use the [icrc1_metadata](https://github.com/dfinity/ICRC-1/tree/main/standards/ICRC-1#metadata_method) method to fetch the metadata entries. All the metadata entries are optional.
```

So they are actually optional, even if in almost all cases they are implemented

Comment (blutooth) [reply src/icrc2-minter/src/tokens/icrc1.rs:127]: I think if metadata is not set, we shouldn't allow bridging.

Comment (veeso) [reply src/icrc2-minter/src/tokens/icrc1.rs:127]: Imo we should stick to the ICRC standard, which suggests but doesn't force to include them. Anyway probably they are set in 99% of the icrc token.

**Review by Maximkaaa — APPROVED** [review]

**Pull request merged**